js/hatrac.js

```diff
@@ -430,13 +431,52 @@ var ERMrest = (function(module) {
             var headers = module.getResponseHeader(response);
             var md5 = headers["content-md5"];
             var length = headers["content-length"];
+            var contentDisposition = headers["content-disposition"];
 
             // If the file is not same, then simply resolve the promise without setting completed and jobDone
             if ((md5 != self.hash.md5_base64) || (length != self.file.size)) {
                 deferred.resolve(self.url);
                 return;
             }
 
+            // hatrac only supports `filename*=UTF-8''` in the content disposition so check for this string to get the filename
+            var contentDispositionPrefix = "filename*=UTF-8''";
+            var filenameIndex = contentDisposition.indexOf(contentDispositionPrefix) + contentDispositionPrefix.length;
+
+            // check if filename in content disposition is different from filename being uploaded
+            // if it is, create an update metadata request for updating the content-disposition
+            if (contentDisposition.substring(filenameIndex, contentDisposition.length) != self.file.name.replace(FILENAME_REGEXP, '_')) {
+                // Prepend the url with server uri if it is relative
+                var url =  self._getAbsoluteUrl(self.url + ";metadata/content-disposition");
+
+                var data = "filename*=UTF-8''" + self.file.name.replace(FILENAME_REGEXP, '_');
+
+                if (!contextHeaderParams || !_isObject(contextHeaderParams)) {
```

Comment:
If `contextHeaderParams` wasn't passed to this function, line 417-423 would create it. So this `if` statement is pointless and this code will always use the same action for both requests. Instead you should assume `contextHeaderParams` is available here and do

```
contextHeaderParams.action = 'upload/metadata/update';
```

So 

```ts
...

var data = "filename*=UTF-8''" + self.file.name.replace(FILENAME_REGEXP, '_');

contextHeaderParams.action = 'upload/metadata/update';
var config = {
    headers: _generateContextHeader(contextHeaderParams)
};
config.headers['content-type'] = 'text/plain';

...
```